leaderboard/src/lib.rs

```diff
@@ -17,15 +17,33 @@ impl HttpServer for LeaderboardActor {
         let path = &req.path[1..req.path.len()];
         let segments: Vec<&str> = path.trim_end_matches('/').split('/').collect();
 
-        match (req.method.as_ref(), segments.as_slice()) {
+        let mut result = match (req.method.as_ref(), segments.as_slice()) {
             ("GET", ["leaderboards"]) => get_leaderboards(ctx).await,
             ("POST", ["leaderboards"]) => create_leaderboard(ctx, deser_json(&req.body)?).await,
             ("GET", ["leaderboards", leaderboard_id]) => get_leaderboard(ctx, leaderboard_id).await,
             ("POST", ["leaderboards", leaderboard_id, "scores"]) => {
                 post_score(ctx, leaderboard_id, deser_json(&req.body)?).await
             }
             (_, _) => Ok(HttpResponse::not_found()),
-        }
+        };
+
+        result
+            .as_mut()
+            .map(|res| {
+                res.header.extend([
+                    (
+                        "Access-Control-Allow-Origin".to_owned(),
+                        vec!["*".to_owned()],
+                    ),
+                    (
+                        "Content-Type".to_owned(),
+                        vec!["application/json".to_owned()],
+                    ),
+                ]);
+            })
+            .unwrap_or_default();
+
+        result
```

Comment:
Nit: I think either of these are more clear 

1) Drop the `mut` from `let result`, then:

```rust
result.map(|mut res| {
    res.header.extend([
        (
            "Access-Control-Allow-Origin".to_owned(),
            vec!["*".to_owned()],
        ),
        (
            "Content-Type".to_owned(),
            vec!["application/json".to_owned()],
        ),
    ]);
    res
})
```

2) Leave the `let mut result`, and add a `?` to the end of the assignment, so we don't have to explicitly handle the error case:
```rust
result.header.extend([
    (
        "Access-Control-Allow-Origin".to_owned(),
        vec!["*".to_owned()],
    ),
    (
        "Content-Type".to_owned(),
        vec!["application/json".to_owned()],
    ),
]);
Ok(result)
```

Reply:
Door number 2!